### PR TITLE
DDT HW

### PR DESCRIPTION
**Pull request by katherinaGo**

### CODE REVIEW

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        LocationInput.SendKeys(searchWord);
+        ChosenCityLine.Click();
+        Waiter.WaitForCondition(SelectedLocationChosen(searchWord).IsElementDisplayedOnPage);
+        return new JobSearchResultPage();
+    }
+
+    public JobSearchResultPage SearchJobSkillsByKeyWord(string searchWord)
+    {
+        SkillsSection.Click();
+        Thread.Sleep(1300); // can't find locator for skillsDropdown panel that it could work without Thread.Sleep
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs]: You can remove "and @aria-hidden='false'" path from SkillsDropdownPanel locator try to use Waiter.WaitForCondition(() => !SkillsDropdownPanel.GetTextFromAttribute("class").Contains("hiden"));

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Core/Elements/BaseElement.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        Element = Browser.Driver.FindElement(locator);
+    }
+
+    protected BaseElement(IWebElement element)
+    {
+        Element = element;
+    }
+
+    public string GetText() => Element.Text.Trim();
+    public string GetTextFromAttribute() => Element.GetAttribute("innerText");
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Core/Elements/BaseElement.cs]: make GetTextFromAttribute() more abstract. (add a parameter to be able to use this method with different attributes)

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Panels/Footer.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    public Link ApplicantPrivacyNotice => new Link(By.XPath("//*[@href='/applicant-privacy-notice']"));
+    public Link WebAccessibility => new Link(By.XPath("//*[@href='/web-accessibility-statement']"));
+
+    public Link LinkedIn => new Link(By.XPath(
+        "//*[@class='footer__social-link' and @href='https://www.linkedin.com/company/epam-systems/']"));
+
+    public Link Twitter => new Link(By.XPath(
+        "//*[@class='footer__social-link' and @href='https://twitter.com/EPAMSYSTEMS']"));
+
+    public Link FaceBook =>
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Panels/Footer.cs]: Just a note. Links can be changed. Therefore, it is not the best solution to explicitly write links in locators.

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Core/Helper/UiTestSettings.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    public static TimeSpan WebDriverTimeOut => TimeSpan.FromSeconds(ProjectSetUpInfo.WebDriverTimeOut);
+    public static BrowserType GetCurrentBrowser => Browser;
+
+    public static string ApplicationUrl => ProjectSetUpInfo.ApplicationUrl;
+    public static string JobListingUrl => ProjectSetUpInfo.JobListingUrl;
+    private static BrowserType Browser => EnumParser.GetEnumValueByDescription<BrowserType>("DefaultBrowser");
+    private static ProjectSetUpInfo ProjectSetUpInfo => GetTestInfoFromJson();
+
+    private static ProjectSetUpInfo GetTestInfoFromJson() => JsonParser.Deserialize<ProjectSetUpInfo>(
+        "/Users/kate/RiderProjects/Test_Automation_Framework/EpamWebSiteTestingFramework/Epam.TestAutomation.TestData/projectsetup.json"
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Core/Helper/UiTestSettings.cs]: write another path to json files

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Panels/Footer.cs (hunk 3 above)
Comment (katherinaGo) [reply EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Panels/Footer.cs:29]: It should be Button?

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Core/Helper/UiTestSettings.cs (hunk 4 above)
Comment (katherinaGo) [reply EpamWebSiteTestingFramework/Epam.TestAutomation.Core/Helper/UiTestSettings.cs]: will be fixed when merging branches (resolving conflicts)

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs (hunk 1 above)
Comment (katherinaGo) [reply EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs]: it helped! thanks!

**Comment by katherinaGo** [comment]
Fixed. Check please.

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Tests/JobListingTests.cs [hunk 5]
```diff
@@ -0,0 +1,95 @@
+using Epam.TestAutomation.Core.Helper;
+using Epam.TestAutomation.Pages.PageObjects.Pages;
+using Epam.TestAutomation.TestData.CareersJobListingsTestInfo;
+
+namespace Epam.TestAutomation.Tests;
+
+// [TestFixture]
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Tests/JobListingTests.cs]: should it be commented?

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        LocationInput.SendKeys(location);
+        ChosenCityLine.Click();
+        SkillsSection.Click();
+        Waiter.WaitForCondition(() => !SkillsDropdownPanel.GetTextFromAttribute("class").Contains("hidden"));
+        SkillOption(skill).Click();
+        FindButton.Click();
+        return new JobSearchResultPage();
+    }
+
+    public JobSearchResultPage FillFiltersWithSearchJobDataToGetErrorMessage(string job, string location)
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs]: It is better to make a more general method for filling fields. Not always a combination of parameters will return an error. You can use FillFiltersWithSearchJobData method. Just add conditions 
if  (job != null) 
{KeywordInputField.SendKeys(job);
        LocationsDropdown.Click();}
and the same for all parameters

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Core/Helper/TestDataSettings.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
+using Epam.TestAutomation.TestData.CareersJobListingsTestInfo;
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Core/Helper/TestDataSettings.cs]: Better to move class to TestData project.

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobSearchResultPage.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        new Label(By.XPath("//*[@class='search-result__error-message' and @role ='alert']"));
+
+    public override bool IsOpened()
+    {
+        return Browser.Driver.GetUrl().Equals(UiTestSettings.JobListingUrl);
+    }
+
+    public override void OpenUrl()
+    {
+        throw new NotImplementedException();
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobSearchResultPage.cs]: Should be implemented

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    public void SearchJobSkillsByKeyWord(string searchWord)
+    {
+        SkillsSection.Click();
+        Waiter.WaitForCondition(() => !SkillsDropdownPanel.GetTextFromAttribute("class").Contains("hidden"));
+        SkillOption(searchWord).Click();
+        FindButton.Click();
+        Waiter.WaitForCondition(ChosenSkillFilter.IsElementDisplayedOnPage);
+    }
+
+    public void FillFiltersWithSearchJobData(string job, string skill, string location)
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs]: you can make all parameters optional (string job = null, string skill = null, string location = null) delete another methods to fill in the fields and use only one method to fill in the search fields.

**Comment by katherinaGo** [comment]
Check please, I've made 2 methods with not mandatory params. 
I can't combine these 2 methods in one unfortunately ( FillFiltersWithValidSearchJobData & FillFiltersWithInValidSearchJobData)

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+        if (!string.IsNullOrEmpty(skill))
+        {
+            SkillsSection.Click();
+            Waiter.WaitForCondition(() => !SkillsDropdownPanel.GetTextFromAttribute("class").Contains("hidden"));
+            SkillOption(skill).Click();
+            Waiter.WaitForCondition(() => ChosenSkillFilter.IsElementDisplayedOnPage());
+        }
+    }
+
+    public void FillFiltersWithInValidSearchJobData(string job = null, string location = null)
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs]: replace the use of this method with FillFiltersWithValidSearchJobData(rename it) and remove.

File: EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+        return Browser.Driver.GetUrl().Equals(UiTestSettings.JobListingUrl);
+    }
+
+    public override void OpenUrl()
+    {
+        Browser.Driver.GotToWebPageUrl(UiTestSettings.JobListingUrl);
+        Waiter.WaitForCondition(() => KeywordInputField.IsElementDisplayedOnPage());
+    }
+
+    public void FillFiltersWithValidSearchJobData(string job = null, string skill = null, string location = null)
```
Comment (MariaEroshevich) [inline EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs]: rename to FillFiltersWithSearchJobData

**Review by MariaEroshevich — APPROVED** [review]

**Pull request merged**